### PR TITLE
Authentication and authorization

### PR DESCRIPTION
**Pull request by CraigJZ**

### CODE REVIEW

**Comment by jcavena** [comment]
Tests aren't passing. Were they passing before?

**Comment by CraigJZ** [comment]
@jcavena I had an issue in my local env that was throwing things off.

**Comment by CraigJZ** [comment]
I made a lot of assumptions in setting this branch up, so I'm definitely open to changes we might want to make in how this is setup.

**Review by jcavena — COMMENTED** [review]
There are some empty .coffee files in this PR. Also users_helper.rb

File: config/initializers/devise.rb [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+  # If 401 status code should be returned for AJAX requests. True by default.
+  # config.http_authenticatable_on_xhr = true
+
+  # The realm used in Http Basic Authentication. 'Application' by default.
+  # config.http_authentication_realm = 'Application'
+
+  # It will change confirmation, password recovery and other workflows
+  # to behave the same regardless if the e-mail provided was right or wrong.
+  # Does not affect registerable.
+  # config.paranoid = true
```
Comment (jcavena) [inline config/initializers/devise.rb]: I recommend we turn this on. This prevents making users enumerable. [`See here`](https://github.com/plataformatec/devise/wiki/How-To:-Using-paranoid-mode,-avoid-user-enumeration-on-registerable)

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update_attributes(user_params)
+      redirect_to users_path
+      flash[:success] = "User updated"
```
Comment (gxespino) [inline app/controllers/users_controller.rb:17]: Is this line evaluated or does it need to be before the redirect?

Comment (jcavena) [reply app/controllers/users_controller.rb:17]: I think it still gets evaluated if there is no explicit return. That's why you can have double render or redirect errors sometimes.

Comment (jcavena) [reply app/controllers/users_controller.rb:17]: I think it technically depends on what the redirect_to evaluates to as well (true/false), but it should typically eval that next line.

Comment (gxespino) [reply app/controllers/users_controller.rb:17]: I didn't get to check as I'm not in my computer. Might be good to add an assertion for the flash messages either way.

Comment (gxespino) [reply app/controllers/users_controller.rb:17]: Actually, just thought about it and you're right. I've definitely had a few double render errors in my day!

Comment (CraigJZ) [reply app/controllers/users_controller.rb:17]: I doubled-checked and it does get evaluated.  If it's preferred to have it before the redirect, I can move it.  Either way, I'll add some assertions for them later on today.  Thanks!

Comment (gxespino) [reply app/controllers/users_controller.rb:17]: Fine as is, your call.

**Review by jcavena — APPROVED** [review]

**Pull request merged**